server/app/app.js

```diff
@@ -0,0 +1,142 @@
+const express = require('express');
+const Promise = require('bluebird');
+var cors = require('cors');
+const app = express();
+const { WishList, WishListItem } = require('../models');
+const errorHandler = require('../middleware/errorHandler');
+
+app.use(express.json());
+app.use(errorHandler);
+app.use(cors());
+
+const getAllWishlists = async () => {
+  return WishList.fetchAll({ withRelated: ['items'], require: true });
+};
+
+const getWishlistById = async (id) => {
+  return WishList.where('id', '=', id).fetch({
+    withRelated: ['items'],
+    require: true,
+  });
+};
+
+const createWishlist = async ({ name, author, items }) => {
+  return new WishList({ name, author })
+    .save(null, {
+      require: true,
+      method: 'insert',
+    })
+    .then(({ id }) => {
+      return Promise.all(
+        items.map((item) => {
+          return new WishListItem({ name: item, wishlist_id: id }).save(null, {
+            require: true,
+            method: 'insert',
+          });
+        })
+      );
+    });
+};
+
+const deleteWishlist = async (id) => {
+  return new WishList({ id }).destroy({ require: true });
+};
+
+const updateWishlist = async (id, { name, author }) => {
+  return new WishList({ id, name, author }).save(null, {
+    require: true,
+    method: 'update',
+  });
+};
+
+const updateItem = async (id, { name }) => {
+  return new WishListItem({ id }).save({ name }, {
+    require: true,
+    method: 'update',
+    patch: true,
+  });
+};
+
+const deleteItem = async (id) => {
+  return new WishListItem({ id }).destroy({ require: true });
+};
+
+app.get('/wishlist', async (req, res) => {
+  getAllWishlists()
+    .then((wishlists) => {
+      res.status(200).send(wishlists);
+    })
+    .catch((error) => {
+      res.status(500).send();
```

Comment:
In situations where we're sending a status code and nothing else, it *may* be preferable to use sendStatus, which will send the string message equivalent of the code as the content (i.e. res.sendStatus(500) is the same as res.status(500).send('Internal Server Error') ).  This allows the client to make easy use of the string for generating an error popup to show the user, etc.

In a similar vein - if we chose NOT to sendStatus, we might consider sending something else back to the client in the case of an error - the error itself, such as:
`.catch((error) => {
      res.status(500).send(error);
    });`

or 
`.catch((error) => {
      res.status(500).send({error});
    });`
to send it in an object.  This is not a NECESSARY change, of course, but can provide some extra information to the client (which, in this case, since we're writing it as well, we can make use of if we choose to create more explicit error handling).

Reply:
I do like the `.sendStatus()` solution. I generally try not to send errors verbatim in any response for security reasons (even though security practices are out of the scope of this project)